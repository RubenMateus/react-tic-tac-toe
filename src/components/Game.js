import React, { useReducer } from 'react';
import { css } from 'emotion';

import Grid from './Grid';

import { reducer, getInitialState } from '../state/reducer';

const Game = () => {
  const [state, dispatch] = useReducer(reducer, getInitialState());
  const { grid, status, turn } = state;

  const handleClick = (x, y) => {
    dispatch({ type: 'CLICK', payload: { x, y } });
  };

  const reset = () => {
    dispatch({ type: 'RESET' });
  };

  return (
    <div
      className={css`
        display: inline-block;
      `}
    >
      <div
        className={css`
          display: flex;
          justify-content: space-between;
        `}
      >
        <div>Next Turn : {turn}</div>
        <div>{status === 'success' ? `${turn} won!` : null}</div>
        <button onClick={reset} type="button">
          Reset
        </button>
      </div>
      <Grid grid={grid} onClick={handleClick} />
    </div>
  );
};

export default Game;
