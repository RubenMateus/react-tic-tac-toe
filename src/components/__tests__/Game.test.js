import React from 'react';
import { render } from '@testing-library/react';
import Game from '../Game';

describe('<Grid />', () => {
  test('renders correctly', () => {
    const { asFragment } = render(<Game />);

    expect(asFragment()).toMatchSnapshot();
  });
});
