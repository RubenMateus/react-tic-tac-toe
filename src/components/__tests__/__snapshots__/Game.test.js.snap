// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Grid /> renders correctly 1`] = `
<DocumentFragment>
  <div
    class="css-1w73uij"
  >
    <div
      class="css-yj3haq"
    >
      <div>
        Next Turn : X
      </div>
      <div />
      <button
        type="button"
      >
        Reset
      </button>
    </div>
    <div
      class="css-1w73uij"
    >
      <div
        class="css-1x1afi9"
      >
        <div
          class="css-ppfkvi"
        >
          <button
            class="css-113hvi"
            type="button"
          />
        </div>
        <div
          class="css-ppfkvi"
        >
          <button
            class="css-113hvi"
            type="button"
          />
        </div>
        <div
          class="css-ppfkvi"
        >
          <button
            class="css-113hvi"
            type="button"
          />
        </div>
        <div
          class="css-ppfkvi"
        >
          <button
            class="css-113hvi"
            type="button"
          />
        </div>
        <div
          class="css-ppfkvi"
        >
          <button
            class="css-113hvi"
            type="button"
          />
        </div>
        <div
          class="css-ppfkvi"
        >
          <button
            class="css-113hvi"
            type="button"
          />
        </div>
        <div
          class="css-ppfkvi"
        >
          <button
            class="css-113hvi"
            type="button"
          />
        </div>
        <div
          class="css-ppfkvi"
        >
          <button
            class="css-113hvi"
            type="button"
          />
        </div>
        <div
          class="css-ppfkvi"
        >
          <button
            class="css-113hvi"
            type="button"
          />
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
